dpctl/tensor/libtensor/source/boolean_reductions.cpp

```diff
@@ -37,10 +37,13 @@
 #include "dpctl4pybind11.hpp"
 
 #include "kernels/boolean_reductions.hpp"
+#include "kernels/dpctl_tensor_types.hpp"
 #include "utils/type_utils.hpp"
 
 namespace py = pybind11;
 
+static_assert(std::is_same_v<py::ssize_t, dpctl::tensor::ssize_t>);
```

Comment:
Is there a specific reason behind the selection of files for these static asserts?

Reply:
The thinking was to make sure that compiling fails in case of type mismatch. I thought `static_assert` should be present in every Python extension module.

Reply:
That makes sense. Would it perhaps be better then to move the asserts to the module `.cpp` files themselves?

So then `static_assert(std::is_same_v<py::ssize_t, dpctl::tensor::ssize_t>);` would appear in `tensor_elementwise.cpp` and `tensor_reductions.cpp` instead?